pyteal/ast/subroutine.py

```diff
@@ -344,20 +453,27 @@ def evaluateSubroutine(subroutine: SubroutineDefinition) -> SubroutineDeclaratio
     - -------- -----
     Type 1 (by-value): these have python type Expr
     Type 2 (by-reference): these have python type ScratchVar
+    Type 3 (ABI): these are ABI typed variables with scratch space storage, and still pass by value
 
     Usage (A) "argumentVars" - Storing pre-placed stack variables into local scratch space:
         Type 1. (by-value) use ScratchVar.store() to pick the actual value into a local scratch space
         Type 2. (by-reference) ALSO use ScratchVar.store() to pick up from the stack
             NOTE: SubroutineCall.__teal__() has placed the _SLOT INDEX_ on the stack so this is stored into the local scratch space
+        Type 3. (ABI) abi_value.stored_value.store() to pick from the stack
 
     Usage (B) "loadedArgs" - Passing through to an invoked PyTEAL subroutine AST:
         Type 1. (by-value) use ScratchVar.load() to have an Expr that can be compiled in python by the PyTEAL subroutine
         Type 2. (by-reference) use a DynamicScratchVar as the user will have written the PyTEAL in a way that satisfies
             the ScratchVar API. I.e., the user will write `x.load()` and `x.store(val)` as opposed to just `x`.
+        Type 3. (ABI) use abi_value itself after storing stack value into scratch space.
     """
 
     def var_n_loaded(param):
```

Comment:
I know you didn't write this, but can you type the argument as a `str`, and the return type as `Tuple[ScratchVar, Union[Expr, ScratchVar, abi.BaseType]]` (or whatever you think is more appropriate)? I think this would help eliminate the need for the new cast you added.